### PR TITLE
Controllers specs

### PR DESCRIPTION
**Pull request by AlaaAlsalem**
- Added Rspec to gem file
- Add Rspec testing files for controllers
- No linter errors

### CODE REVIEW

File: spec/requests/users_spec.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+require 'rails_helper'
+
+RSpec.describe 'Users', type: :request do
+  describe 'GET /users' do
+    it 'works! (now write some real specs)' do
+      get users_index_path
+      expect(response).to have_http_status(200)
+    end
+  end
+end
```
Comment (FrederickMih) [inline spec/requests/users_spec.rb]: - [ ] As stated in the requirements of this project, both the index and show actions should be tested. Spec files include:
```
describe 'GET /index' do
  ...
end

describe 'GET /show' do
  ...
end
```

- Tests that ensure the correct view template is rendered
- Tests that ensure the response body includes the correct placeholder text
Let's kindly meet up with the requirements.

File: spec/requests/posts_spec.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
+require 'rails_helper'
+
+describe Post, type: :request do
```
Comment (FrederickMih) [inline spec/requests/posts_spec.rb:3]: - [ ] Let's once more kindly make sure all tests cases are passing:
Screenshot from 2023-06-24 16-07-14

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand: :raising_hand: @AlaaAlsalem , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!





## Highlights :green_circle: 

- Good testing practices: Tests group into describe blocks :+1: 


## Required Changes ♻️
**N/A**
_Check the comments under the review._

## Optional suggestions
**N/A**
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/users_spec.rb (hunk 1 above)
Comment (rloterh) [reply spec/requests/users_spec.rb]: well resolved! ✔️

File: spec/requests/posts_spec.rb (hunk 2 above)
Comment (rloterh) [reply spec/requests/posts_spec.rb:3]: well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Aladdin,  

Your project is complete! 🏁 There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 🎯 
🎲 Good job on correctly implementing request specs!

## Optional suggestions 🧰 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

**You can also consider:**
Kindly include instructions in your ReadMe on how to run the test cases of your project, under the _Getting Started_ section.
Implementing the above change(s) will make your ReadMe/ repository stand out even better! 😉

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**